Development/v0.1.5

amoshydra opened the pull request.

tayjiehao committed:
Add webworkify for markdown-it processor

amoshydra committed:
Merge branch 'development/v0.1.4' into development/webworkify-webpack

amoshydra committed:
Fix typo

jiayingy committed:
Fix width of view type buttons

amoshydra committed:
Merge pull request #385 from nus-mtp/ui/view-type-width

Fix width of view type buttons

jiayingy committed:
Fix double click compatibility with iOS

jiayingy committed:
Fix double click compatibility with iOS

jiayingy committed:
Remove console log line

jiayingy committed:
Reduce onclick timeout to 1s

jiayingy committed:
Check for onClick x,y position before executing double click

jiayingy committed:
Update clicked coordinates when another click event is received

amoshydra committed:
Add feature list at /features

chuajiaxuan committed:
Update feature.txt with latest version

chuajiaxuan committed:
Merge pull request #387 from nus-mtp/dev/add-features-at-router

Add feature list at /features

chuajiaxuan committed:
Fix typo in features.txt

tharain committed:
Merge remote-tracking branch 'refs/remotes/origin/development/v0.1.5' into database_v.1.5

tayjiehao committed:
Implement promise worker

tayjiehao committed:
Merge branch 'development/webworkify-webpack' into dev/merge-dev-with-webworker

jiayingy committed:
Replace checking of exact clicked coordinates with threshold

chuajiaxuan committed:
Refactor diagramsRenderer Component to improve rendering performance

tayjiehao committed:
Fix an issue with weborkify throwing an error on initilization

amoshydra committed:
Implement support for pagination post process fns

chuajiaxuan committed:
Refactor code, resolve bugs for diagramsRenderer

chuajiaxuan committed:
Update test case for diagramsRenderer

chuajiaxuan committed:
Resolve styling issue in print mode

chuajiaxuan committed:
Fix diagrams width css issues

amoshydra committed:
Update documentation for PageRenderer

amoshydra committed:
Merge pull request #386 from nus-mtp/ui/fix-for-ios

Ui/fix for ios

amoshydra committed:
Remove additional space

amoshydra committed:
Add line number injection plugin

amoshydra committed:
Allow injected line number into the html

amoshydra committed:
Update test case for #389

amoshydra committed:
Implement scroll detection and mover on editor

amoshydra committed:
Implement editor to viewer scrolling

amoshydra committed:
Fix element utils documentation

amoshydra committed:
Extract scroll function in to element utils

amoshydra committed:
Manually convert number string into integer first before comparison

amoshydra committed:
Update scrolling code from Pawel Grzybek

amoshydra committed:
Move scroll check logic into a new scrollSync component

chuajiaxuan committed:
Merge pull request #390 from nus-mtp/dev/update-test-case-for-#389

Update test case for #389

amoshydra committed:
Fix typo in event message

amoshydra committed:
Change newLinePosition calculation offset

- The previous 2 offset will cause line 1 to never be reached

amoshydra committed:
Implement 2 ways scroll sync

jiayingy committed:
Emit focus event when click event is emitted

jiayingy committed:
Fix margin of buttons on iphone se view

jiayingy committed:
Animate list icons to slide in from left

amoshydra committed:
Remove support for other easing animation

amoshydra committed:
Seperate editor and viewer scrolling event, dynamically toggle scroll watch event

amoshydra committed:
Remove unused console.log

amoshydra committed:
Change editor scroll checking to be more robust

amoshydra committed:
Update search algorithm to scan every pixel from the top

amoshydra committed:
Use better codemirror api for retreive height

amoshydra committed:
Reduce viewer side scroll condition checking

amoshydra committed:
Offset element search by 50px from the top

amoshydra committed:
Remove unused dependency for editor

amoshydra committed:
Remove unused console.log

amoshydra committed:
Make attachment of scrollSync simpler on viewer component

- vue watcher and event listener attachment can now be done inside scrollSync library
- any component that wish to make use of the scrollSync capability just need to add setDomBehaviour

amoshydra committed:
Expand elementUtils capability to get Window and documents

amoshydra committed:
Change scrollTo to use getBoundingClientRect instead of relying on offsetTop

- This is done so that, it could accomondate more scanerio such as when the element has been transformed.

amoshydra committed:
Fix error where 'this' is unresolved in scrollSync vueHelper

amoshydra committed:
Apply scrollSync to pages and slides mode

amoshydra committed:
Remove debug console

amoshydra committed:
Merge pull request #389 from nus-mtp/dev/inject-line-number-into-md-html

Inject line number into the rendered markdown HTML

amoshydra committed:
Merge pull request #388 from nus-mtp/dev/mermaidAPI-enhancement

Refactor diagramsRenderer Component to improve rendering performance

amoshydra committed:
Merge branch 'development/v0.1.5' of https://github.com/nus-mtp/sashimi-note into dev/implement-scroll-sync

tharain committed:
Update constants.js to new feature of text file and number permission

tharain committed:
Add default markdown file to initDataGenerator.js

tharain committed:
Update dateTime to long time for dataAdd.js

tharain committed:
remove checking of starting content for creating new file

tharain committed:
update sql array to handle multiple items

tharain committed:
Add default database name to query.js for testing

tharain committed:
Remove import of exceptions in storage.js

not used

tharain committed:
Add test cases for entitiesCreator

tharain committed:
Add test cases for initDataGenerator

tharain committed:
add brackets for doneAfter in tableCreator.spec.js

tharain committed:
Add test cases for dataAdd

tharain committed:
Add test cases for query.spec.js

tharain committed:
Add test cases for sqlCommands

tharain committed:
Update comments to storage.js and remove deleteAll function

tharain committed:
Add test case for multiple creation to sqlArray.spec.js

tharain committed:
Fix test case error to sqlCommands.spec.js

tharain committed:
Fix bug of not adding last item to sqlArray.js

tharain committed:
fix all bugs during test for sqlCommands.js

amoshydra committed:
Replace array.includes to generic for loop

amoshydra committed:
Merge pull request #391 from nus-mtp/ui/fix-for-ios

Ui/fix for ios

jiayingy committed:
Resolve merge conflict

amoshydra committed:
Fix typo

jiayingy committed:
Handle rename error for folder

jiayingy committed:
Remove unused style binding

jiayingy committed:
Remove unused codes

amoshydra committed:
Merge pull request #394 from nus-mtp/database_v.1.5

Database v.1.5 - added test cases, feature add and bugs fixed

amoshydra committed:
Merge branch 'development/v0.1.5' into dev/implement-scroll-sync

amoshydra committed:
Fix typo

amoshydra committed:
Merge pull request #377 from nus-mtp/development/webworkify-webpack

Add webworkify for markdown-it processor

amoshydra committed:
Update keyframes.scss

jiayingy committed:
Merge pull request #396 from nus-mtp/ui/handle-errors-from-filemanager

Ui/handle errors from filemanager

amoshydra committed:
Merge pull request #393 from nus-mtp/dev/implement-scroll-sync

Implement scroll sync for viewer and editor

chuajiaxuan committed:
Move markdown-it-asciimath plugins to plugins folder

chuajiaxuan committed:
Move markdown-it-diagrams to plugins folder

chuajiaxuan committed:
Move conditionalProcessor to plugins folder

chuajiaxuan committed:
Update custom plugin directory location in markdownProcessor

chuajiaxuan committed:
Update custom plugin directory location in Navbar.vue

chuajiaxuan committed:
Resolve eslint error in Html.vue

chuajiaxuan committed:
Update mermaidAPI.min.js to latest version

chuajiaxuan committed:
Update helper function in diagramsRenderer test case

chuajiaxuan committed:
Update diagramsRenderedOutput.txt

chuajiaxuan committed:
Update mermaidOutput.txt

chuajiaxuan committed:
Debug errors shown on travis

chuajiaxuan committed:
Resolve error in travis

chuajiaxuan committed:
Debug errors shown on travis for drawing all diagrams

chuajiaxuan committed:
Resolve errors shown by Travis

amoshydra committed:
Merge pull request #397 from nus-mtp/dev/update-mermaid-dependency

Update mermaid dependency

amoshydra committed:
Merge pull request #398 from nus-mtp/dev/reorganise-plugins

Reorganise plugins and update mermaid dependency

tharain committed:
fix a tiny comment typo

tharain commented on sashimi-webapp/src/assets/styles/keyframes.scss:
```diff
@@ -21,7 +21,20 @@
 @keyframes docListView{
   0% { transform: scale(1.05) }
   100% { transform: scale(1) }
-}
```
eslint new line here

jiayingy committed:
Add throttle function to markdown parsing

amoshydra committed:
Merge pull request #400 from nus-mtp/mini-comment-fix

fix a tiny comment typo

amoshydra committed:
Merge pull request #401 from nus-mtp/dev/add-throttle-function-to-markdown-parsing

Add throttle function to markdown parsing

tharain commented on sashimi-webapp/static/vendors/diagramRendering-fix.css:
```diff
@@ -0,0 +1,6 @@
+pre.graphviz > svg,
+pre.mermaid > svg {
+  width: 100%
+}
```
eslint new line here

jiayingy committed:
Remove unused router import

- it was previously thought to resolve vue router render error message
- however it did not actually resolve the problem

jiayingy committed:
Ignore necessary code that failed eslint check

jiayingy committed:
Remove unused floater component

jiayingy committed:
Resolve eslint error

amoshydra committed:
Merge pull request #402 from nus-mtp/fix/resolve-all-eslint-problems

Resolve all eslint problems in v0.1.5 before merging to master

chuajiaxuan reviewed (commented):
Other than the 2 eslint warning/errors that daphne mentioned, looks good!
